NullPointerException Failed to execute ScheduledFutureTask InvalidationMetaDataFetcher repairUuids

in test

https://hazelcast-l337.ci.cloudbees.com/view/kill/job/kill-members/3/console

is see this logged on the client side.

```
Sep 06, 2018 10:53:48 AM com.hazelcast.client.spi.ClientExecutionService
SEVERE: hz.client_0 [HZ] [3.11-SNAPSHOT] Failed to execute java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@2dae9baf
java.lang.NullPointerException
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.repairUuids(InvalidationMetaDataFetcher.java:139)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.processMemberMetadata(InvalidationMetaDataFetcher.java:126)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.fetchMetadata(InvalidationMetaDataFetcher.java:77)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.runAntiEntropyIfNeeded(RepairingTask.java:156)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.run(RepairingTask.java:125)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:473)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:178)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:292)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```

```
 <connection-strategy async-start="true" reconnect-mode="ASYNC">
        <connection-retry enabled="true">
            <initial-backoff-millis>333</initial-backoff-millis>
            <max-backoff-millis>30000</max-backoff-millis>
            <multiplier>2</multiplier>
        </connection-retry>
    </connection-strategy>
```


```
Sep 06, 2018 10:53:48 AM com.hazelcast.client.map.impl.nearcache.invalidation.ClientMapInvalidationMetaDataFetcher
WARNING: hz.client_0 [HZ] [3.11-SNAPSHOT] Can't extract invalidation meta-data of Member [10.0.0.30]:5701 - 3abd29df-3336-4398-b69a-6d5d877cbaea
java.util.concurrent.ExecutionException: com.hazelcast.client.HazelcastClientOfflineException: Client is offline.
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:109)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:33)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:178)
	at com.hazelcast.client.map.impl.nearcache.invalidation.ClientMapInvalidationMetaDataFetcher.extractMemberMetadata(ClientMapInvalidationMetaDataFetcher.java:70)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.processMemberMetadata(InvalidationMetaDataFetcher.java:117)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.fetchMetadata(InvalidationMetaDataFetcher.java:77)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.runAntiEntropyIfNeeded(RepairingTask.java:156)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.run(RepairingTask.java:125)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:473)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:178)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:292)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
Caused by: com.hazelcast.client.HazelcastClientOfflineException: Client is offline.
	at com.hazelcast.client.connection.nio.DefaultClientConnectionStrategy.beforeGetConnection(DefaultClientConnectionStrategy.java:58)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.checkAllowed(ClientConnectionManagerImpl.java:329)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.getConnection(ClientConnectionManagerImpl.java:304)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.getOrTriggerConnect(ClientConnectionManagerImpl.java:295)
	at com.hazelcast.client.spi.impl.SmartClientInvocationService.getOrTriggerConnect(SmartClientInvocationService.java:73)
	at com.hazelcast.client.spi.impl.SmartClientInvocationService.invokeOnTarget(SmartClientInvocationService.java:68)
	at com.hazelcast.client.spi.impl.ClientInvocation.invokeOnSelection(ClientInvocation.java:165)
	at com.hazelcast.client.spi.impl.ClientInvocation.invoke(ClientInvocation.java:146)
	at com.hazelcast.client.map.impl.nearcache.invalidation.ClientMapInvalidationMetaDataFetcher.fetchMetadataOf(ClientMapInvalidationMetaDataFetcher.java:62)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.fetchMembersMetadataFor(InvalidationMetaDataFetcher.java:99)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.fetchMetadata(InvalidationMetaDataFetcher.java:73)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.runAntiEntropyIfNeeded(RepairingTask.java:156)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.run(RepairingTask.java:125)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:473)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:178)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:292)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:96)
	... 16 more

Sep 06, 2018 10:53:48 AM com.hazelcast.client.spi.ClientExecutionService
SEVERE: hz.client_0 [HZ] [3.11-SNAPSHOT] Failed to execute java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@2dae9baf
java.lang.NullPointerException
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.repairUuids(InvalidationMetaDataFetcher.java:139)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.processMemberMetadata(InvalidationMetaDataFetcher.java:126)
	at com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher.fetchMetadata(InvalidationMetaDataFetcher.java:77)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.runAntiEntropyIfNeeded(RepairingTask.java:156)
	at com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.run(RepairingTask.java:125)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:473)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:178)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:292)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```

--- a/hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/nearcache/invalidation/ClientCacheInvalidationMetaDataFetcher.java
+++ b/hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/nearcache/invalidation/ClientCacheInvalidationMetaDataFetcher.java
@@ -16,7 +16,6 @@
 
 package com.hazelcast.client.cache.impl.nearcache.invalidation;
 
-
 import com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl;
 import com.hazelcast.client.impl.protocol.ClientMessage;
 import com.hazelcast.client.impl.protocol.codec.CacheFetchNearCacheInvalidationMetadataCodec.ResponseParameters;

--- a/hazelcast-client/src/main/java/com/hazelcast/client/map/impl/nearcache/invalidation/ClientMapInvalidationMetaDataFetcher.java
+++ b/hazelcast-client/src/main/java/com/hazelcast/client/map/impl/nearcache/invalidation/ClientMapInvalidationMetaDataFetcher.java
@@ -16,7 +16,6 @@
 
 package com.hazelcast.client.map.impl.nearcache.invalidation;
 
-
 import com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl;
 import com.hazelcast.client.impl.protocol.ClientMessage;
 import com.hazelcast.client.impl.protocol.codec.MapFetchNearCacheInvalidationMetadataCodec.ResponseParameters;

--- a/hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/InvalidationMetaDataFetcher.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/InvalidationMetaDataFetcher.java
@@ -16,7 +16,6 @@
 
 package com.hazelcast.internal.nearcache.impl.invalidation;
 
-import com.hazelcast.core.HazelcastInstanceNotActiveException;
 import com.hazelcast.core.Member;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.nio.Address;
@@ -97,12 +96,8 @@ public abstract class InvalidationMetaDataFetcher {
             Address address = member.getAddress();
             try {
                 futureByMember.put(member, fetchMetadataOf(address, names));
-            } catch (HazelcastInstanceNotActiveException e) {
-                logger.finest(e);
             } catch (Exception e) {
-                if (logger.isWarningEnabled()) {
-                    logger.warning(format("Can't fetch invalidation meta-data of %s", member), e);
-                }
+                handleExceptionWhileProcessingMetadata(member, e);
             }
         }
 
@@ -115,16 +110,23 @@ public abstract class InvalidationMetaDataFetcher {
         MetadataHolder resultHolder = new MetadataHolder();
         try {
             extractMemberMetadata(member, future, resultHolder);
-        } catch (HazelcastInstanceNotActiveException e) {
-            logger.finest(e);
         } catch (Exception e) {
-            if (logger.isWarningEnabled()) {
-                logger.warning(format("Can't extract invalidation meta-data of %s", member), e);
-            }
+            handleExceptionWhileProcessingMetadata(member, e);
+            return;
         }
 
         repairUuids(resultHolder.partitionUuidList, handlers);
         repairSequences(resultHolder.namePartitionSequenceList, handlers);
+    }
+
+    protected void handleExceptionWhileProcessingMetadata(Member member, Exception e) {
+        if (e instanceof IllegalStateException) {
+            // log at finest when
+            // HazelcastInstanceNotActive, HazelcastClientNotActive or HazelcastClientOffline exception
+            logger.finest(e);
+        } else if (logger.isWarningEnabled()) {
+            logger.warning(format("Can't fetch or extract invalidation meta-data of %s", member), e);
+        }
     }
 
     private List<String> getDataStructureNames(ConcurrentMap<String, RepairingHandler> handlers) {

